### PR TITLE
Fixing crazy blinds on MQTT auto discover blinds types

### PR DESCRIPTION
**Pull request by CyrilB1531**
Fixing #5217 on MQTT auto discover blinds devices:
- Adding sTypeBlindsT21 as generic MQTT blinds
- Setting MQTT Autodiscover code to match generic blinds
- Changing WebServer to send good values to web site and good action to MQTT Auto discover
- Changing LightController and DashboardController to reflect changes

Tested with a Tasmota dual R3 as blind switch on Compact, Normal and Mobile Dashboard of web site

### CODE REVIEW

**Comment by gizmocuz** [comment]
Thank you very much for this PR!
After a quick check, there is one problem ... **'#define sTypeBlindsT21 0x15	//Generic'** in RFXtrx.h, this file is from the manufactorer and will be overwritten by new updates.
Could you add a new blinds in 'RFXNames.h' ? (Maybe label it sTypeBlindsMQTT)

**Comment by CyrilB1531** [comment]
Thanks for your feedback, 
If I add new blinds, I will add:
	STYPE_BlindsMQTT,
	STYPE_BlindsInvertedMQTT,
	STYPE_BlindsPercentageMQTT,
	STYPE_BlindsPercentageInvertedMQTT,
	STYPE_BlindsPercentageWithStopMQTT,
	STYPE_BlindsPercentageInvertedWithStopMQTT,

May be I can use Hardware HwdType instead. I check that

**Comment by gizmocuz** [comment]
@CyrilB1531 , I do not think we need any additional blind types as you know the hardware is MQTT correct?
So obviously in the MQTT class you do not need them, and for the eventsystem you can check the hardware type
There should also be no difference in behaviour ?

File: main/RFXtrx.h [hunk 1]
```diff
@@ -881,10 +881,9 @@ SDK version 4.9
 #define sTypeBlindsT18 0x12	//Cherubini
 #define sTypeBlindsT19 0x13	//Louvolite vertical
 #define sTypeBlindsT20 0x14	//Ozroll E-Trans
-#define sTypeBlindsT21 0x15	//Generic
 
-#define blinds_sOpen 0x0
-#define blinds_sClose 0x1
+#define blinds_sClose 0x0
```
Comment (gizmocuz) [inline main/RFXtrx.h]: You can not modify anything in this file

Comment (CyrilB1531) [reply main/RFXtrx.h]: Done, I've reverted my changes

**Comment by gizmocuz** [comment]
Thank you for all the changes and time!!

One question, in the lightscontroller.js i noticed:

```
if(item.HardwareTypeVal != 125)
								xhtm += '\t      <td id="type">' + item.Type + ', ' + item.SubType + ', ' + item.SwitchType;
							else
								xhtm += '\t      <td id="type">' + item.Type + ', ' + item.SwitchType;

```

It is a bit scary to check against 125.... But I did not see this in the DashboardController.... Is there a reason to remove the display of the subtype ?

**Comment by CyrilB1531** [comment]
I've just remove it, it was when MQTT AD Covers was set as blind Type.

**Comment by gizmocuz** [comment]
@CyrilB1531 , thank you very much for this patch and work!! (And also to all testers!!)

**Comment by CyrilB1531** [comment]
Thank you all

**Comment by gizmocuz** [comment]
@CyrilB1531 , is it possible to have a look at this issue:

https://github.com/domoticz/domoticz/issues/5282

Maybe it's because of the changes in the web interface? Seems like adding a virtual blind (dummy hardware) is not working , and I suspect other (real) hardware as well.
Maybe we could keep the web interface the same (when there are no issues with it of course) and patch the code in MQTT AD ?

**Comment by CyrilB1531** [comment]
I Gizmocuz,

Of course I can and I will check it


De : gizmocuz ***@***.***>
Envoyé : samedi 23 juillet 2022 08:07
À : domoticz/domoticz ***@***.***>
Cc : CyrilB1531 ***@***.***>; Mention ***@***.***>
Objet : Re: [domoticz/domoticz] Fixing crazy blinds on MQTT auto discover blinds types (PR #5260)


@CyrilB1531<https://github.com/CyrilB1531> , is it possible to have a look at this issue:

#5282<https://github.com/domoticz/domoticz/issues/5282>

Maybe it's because of the changes in the web interface? Seems like adding a virtual blind (dummy hardware) is not working , and I suspect other (real) hardware as well.
Maybe we could keep the web interface the same (when there are no issues with it of course) and patch the code in MQTT AD ?

**Comment by kiddigital** [comment]
@CyrilB1531 , would it be possible to maybe add some tests? See the test folder.

Even just writing the feature files, without actual implementation, would already be helpful so people can read the intended behavior (it acts as documentation at least).

If you need help or have questions, feel free to use our slack channels. Thx.

**Comment by gizmocuz** [comment]
@CyrilB1531 , this PR is causing the light/dimmer slider on the dashboard to not update anymore when a new value is introduced
A quick reproduction is two add a dummy light/dimmer and open two browsers, one on the switches tab and one on the dashboard
Next set a level on the switches tab

If I check for both _slider and #slider then it works ... but is there a need to introduce a _slider ? all was #slider before and there was no need to check for existence of both

**Comment by CyrilB1531** [comment]
@gizmocuz hello,
if, you've got more than one dimmer device the previous code creates devices with the same id="slider" and when you change the value of one of these devices, all slider change their values. That's why I've changed it. I'll check it to reproduce your issue and fix it then

**Comment by gizmocuz** [comment]
@CyrilB1531 , thanks for checking. Yes you would expect id="123 slider" not "slider"

**Pull request merged**